scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -938,6 +949,29 @@ def apply_filters(self, **kwargs):
                 # handle quarantined tests
                 self.handle_quarantined_tests(instance, plat)
 
+                # Check if scripting is available
+                if self.scripting:
+                    matched_scripting = self.scripting.get_matched_scripting(instance.testsuite.id, plat.name)
+                    if matched_scripting:
+                        # Define a function to validate if the platform is supported by the matched scripting
+                        def validate_boards(platform_scope, platform_from_yaml):
+                            return any(board in platform_scope for board in platform_from_yaml)
+
+                        # Define the types of scripts we are interested in
+                        script_types = {'pre_script': 'pre_script','post_flash_script': 'post_flash_script', 'post_script': 'post_script' }
```

Comment:
why it needs a dictionary with keys == values, not just a list ?

Reply:
updated, now added timeout which will use dict. 
Before it was part of old code with you pointed